docs/upgrade/xenial_upgrade_in_place.rst

```diff
@@ -0,0 +1,317 @@
+ cd ~/Persistent/securedrop                                                     
+ git checkout 0.12.0                                                            
+ git tag -v 0.12.0                                                              
+                                                                                
+You should see ``Good signature from "SecureDrop Release Signing Key"`` in the 
+output of that last command, along with the fingerprint 
+``"2224 5C81 E3BA EB41 38B3 6061 310F 5612 00F4 AD77"``
+                                                                                
+.. caution::                                                                    
+                                                                                
+ If you do not, signature verification has failed and you should not proceed 
+ with the installation. If this happens, please contact us at 
+ securedrop@freedom.press.
+                                                                                
+If the command above returns the expected value, you may proceed with the installation.
+
+In the terminal, run the following command to set up the SecureDrop 
+admin environment:
+
+.. code:: sh
+
+  ./securedrop-admin setup
+
+Next, verify that the SecureDrop configuration matches expected values, by 
+stepping through the configuration using:
+
+.. code:: sh
+
+  ./securedrop-admin sdconfig
+
+Finally, install the Ubuntu 16.04 version of the server application code and
+configuration:
+
+.. code:: sh
+
+  ./securedrop-admin install
+
+You will be prompted for the admin user's passphrase on the servers. Type it in
+and press Enter.
+
+Step 4: Validate the Instance
+-----------------------------
+
+While we have extensively tested the upgrade on recommended hardware, we
+recommend performing the following tests yourself to identify potential issues
+specific to your system configuration.
+
+Validate the kernel version
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Ensure you are logged out, and then type the commands ``ssh app uname -r`` and
+``ssh mon uname -r`` in your terminal window.
+
+The output for both commands should be ``4.4.167-grsec``, which indicates that
+the latest available kernel for SecureDrop is installed on your *Application
+Server* and your *Monitor Server*.
+
+Validate the application version
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To confirm that you are running SecureDrop 0.12.0 for Xenial, on the Tails
+desktop, you should find a shortcut called **SecureDrop Source Interface**.
+Double-click it to launch the Tor browser.
+
+After the *Source Interface* loads, add the path ``/metadata`` to the URL in
+your address bar. If your *Source Interface* can be found at
+``examplenot4real.onion``, then the address you should visit is
+``examplenot4real.onion/metadata``. That page should show you key/value pairs,
+including ``0.12.0`` for ``sd_version`` and ``16.04`` for ``server_os``.
+
```

Comment:
Ensuring OSSEC emails are flowing would be another interesting test to validate the upgrade

Reply:
added a section for this.